docs/data/date-pickers/custom-field/custom-behavior/MaskedMaterialTextField.tsx

```diff
@@ -0,0 +1,168 @@
+import * as React from 'react';
+import dayjs, { Dayjs } from 'dayjs';
+import { useRifm } from 'rifm';
+import TextField from '@mui/material/TextField';
+import useControlled from '@mui/utils/useControlled';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import {
+  DatePicker,
+  DatePickerProps,
+  DatePickerFieldProps,
+} from '@mui/x-date-pickers/DatePicker';
+import { useSplitFieldProps, useParsedFormat } from '@mui/x-date-pickers/hooks';
+import { useValidation, validateDate } from '@mui/x-date-pickers/validation';
+
+const MASK_USER_INPUT_SYMBOL = '_';
+const ACCEPT_REGEX = /[\d]/gi;
+
+const staticDateWith2DigitTokens = dayjs('2019-11-21T22:30:00.000');
+const staticDateWith1DigitTokens = dayjs('2019-01-01T09:00:00.000');
+
+function getValueStrFromValue(value: Dayjs | null, format: string) {
+  if (value == null) {
+    return '';
+  }
+
+  return value.isValid() ? value.format(format) : '';
+}
+
+function MaskedField(props: DatePickerFieldProps<Dayjs>) {
+  const { slots, slotProps, ...other } = props;
+
+  const { forwardedProps, internalProps } = useSplitFieldProps(other, 'date');
+
+  const {
+    format,
+    value: valueProp,
+    defaultValue,
+    onChange,
+    timezone,
+    onError,
+  } = internalProps;
+
+  const [value, setValue] = useControlled({
+    controlled: valueProp,
+    default: defaultValue ?? null,
+    name: 'MaskedField',
+    state: 'value',
+  });
+
+  // Control the input text
+  const [valueStr, setValueStr] = React.useState<string>(() =>
```

Comment:
WDYT about avoiding the abbreviations? 🤔 
Maybe we could use `valueString` or `valueText`?

Reply:
Updated